lib/content-services/src/lib/common/services/saved-searches.service.ts

```diff
@@ -0,0 +1,138 @@
+                        }
+                        return throwError(error);
+                    })
+                );
+            })
+        );
+    }
+
+    saveSearch(newSaveSearch: Pick<SavedSearch, 'name' | 'description' | 'encodedUrl'>): Observable<NodeEntry> {
+        return this.getSavedSearches().pipe(
+            take(1),
+            switchMap((savedSearches: Array<SavedSearch>) => {
+                const updatedSavedSearches = [...savedSearches, { ...newSaveSearch, order: savedSearches.length }];
+                return from(this.nodesApi.updateNodeContent(this.savedSearchFileNodeId, JSON.stringify(updatedSavedSearches))).pipe(
+                    tap(() => this.savedSearches$.next(updatedSavedSearches))
+                );
+            })
+        );
+    }
+
+    getSavedSearchesNodeId(): Observable<string> {
+        let savedSearchesNodeId = localStorage.getItem(this.SAVED_SEARCHES_NODE_ID) ?? '';
+        if (savedSearchesNodeId === '') {
+            return from(this.nodesApi.getNode('-my-')).pipe(
+                first(),
+                map((node) => node.entry.id),
+                concatMap((parentNodeId) =>
+                    from(
+                        this.searchApi.search({
+                            query: {
+                                language: SEARCH_LANGUAGE.AFTS,
+                                query: `cm:name:"saved-searches.json" AND PARENT:"${parentNodeId}"`
+                            }
+                        })
+                    ).pipe(
+                        first(),
+                        concatMap((searchResult: ResultSetPaging) => {
+                            if (searchResult.list.entries.length > 0) {
+                                savedSearchesNodeId = searchResult.list.entries[0].entry.id;
+                                localStorage.setItem(this.SAVED_SEARCHES_NODE_ID, savedSearchesNodeId);
+                            } else {
+                                return this.createSavedSearchesNode(parentNodeId).pipe(
+                                    first(),
+                                    map((node) => {
+                                        localStorage.setItem(this.SAVED_SEARCHES_NODE_ID, node.entry.id);
+                                        return node.entry.id;
+                                    })
+                                );
+                            }
+                            return savedSearchesNodeId;
+                        })
+                    )
+                )
+            );
+        } else {
+            return of(savedSearchesNodeId);
+        }
+    }
+    private createSavedSearchesNode(parentNodeId: string): Observable<NodeEntry> {
+        return from(
+            this.nodesApi.createNode(parentNodeId, {
+                name: 'saved-searches.json',
+                nodeType: 'cm:content'
+            })
+        );
+    }
+
+    private mapFileContentToSavedSearches(blob: Blob): Promise<Array<SavedSearch>> {
```

Comment:
I'm getting this warning (I guess it's because it returns Promise): 
<img width="1083" alt="image" src="https://github.com/user-attachments/assets/874039f4-173a-4f5c-99b1-a6e767415f15">